Traceback (most recent call last):
  File "/Users/juliehartley/Library/Python/3.9/lib/python/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/juliehartley/Library/Python/3.9/lib/python/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/juliehartley/Library/Python/3.9/lib/python/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/Users/juliehartley/Library/Python/3.9/lib/python/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/Users/juliehartley/Library/Python/3.9/lib/python/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/Users/juliehartley/Library/Python/3.9/lib/python/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/juliehartley/Library/Python/3.9/lib/python/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import keras

# Create the model 
model = keras.Sequential()
model.add(keras.layers.Dense(units = 25, activation = 'tanh', input_dim=1))
model.add(keras.layers.Dense(units = 25, activation = 'tanh'))
model.add(keras.layers.Dense(units = 1, activation = 'tanh'))
model.compile(loss='mse', optimizer="adam")

model.fit(X_train,y_train,epochs=200)
y_nn = model.predict(X_test)
print(rmse(y_nn, y_test))
plt.scatter(X_test, y_nn)
plt.scatter(X_test,y_test)
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mModuleNotFoundError[0m                       Traceback (most recent call last)
Cell [0;32mIn [12], line 1[0m
[0;32m----> 1[0m [38;5;28;01mimport[39;00m [38;5;21;01mkeras[39;00m
[1;32m      3[0m [38;5;66;03m# Create the model [39;00m
[1;32m      4[0m model [38;5;241m=[39m keras[38;5;241m.[39mSequential()

File [0;32m~/Library/Python/3.9/lib/python/site-packages/keras/__init__.py:20[0m
[1;32m      1[0m [38;5;66;03m# Copyright 2015 The TensorFlow Authors. All Rights Reserved.[39;00m
[1;32m      2[0m [38;5;66;03m#[39;00m
[1;32m      3[0m [38;5;66;03m# Licensed under the Apache License, Version 2.0 (the "License");[39;00m
[0;32m   (...)[0m
[1;32m     13[0m [38;5;66;03m# limitations under the License.[39;00m
[1;32m     14[0m [38;5;66;03m# ==============================================================================[39;00m
[1;32m     15[0m [38;5;124;03m"""Implementation of the Keras API, the high-level API of TensorFlow.[39;00m
[1;32m     16[0m 
[1;32m     17[0m [38;5;124;03mDetailed documentation and user guides are available at[39;00m
[1;32m     18[0m [38;5;124;03m[keras.io](https://keras.io).[39;00m
[1;32m     19[0m [38;5;124;03m"""[39;00m
[0;32m---> 20[0m [38;5;28;01mfrom[39;00m [38;5;21;01mkeras[39;00m [38;5;28;01mimport[39;00m distribute
[1;32m     21[0m [38;5;28;01mfrom[39;00m [38;5;21;01mkeras[39;00m [38;5;28;01mimport[39;00m models
[1;32m     22[0m [38;5;28;01mfrom[39;00m [38;5;21;01mkeras[39;00m[38;5;21;01m.[39;00m[38;5;21;01mengine[39;00m[38;5;21;01m.[39;00m[38;5;21;01minput_layer[39;00m [38;5;28;01mimport[39;00m Input

File [0;32m~/Library/Python/3.9/lib/python/site-packages/keras/distribute/__init__.py:18[0m
[1;32m      1[0m [38;5;66;03m# Copyright 2019 The TensorFlow Authors. All Rights Reserved.[39;00m
[1;32m      2[0m [38;5;66;03m#[39;00m
[1;32m      3[0m [38;5;66;03m# Licensed under the Apache License, Version 2.0 (the "License");[39;00m
[0;32m   (...)[0m
[1;32m     13[0m [38;5;66;03m# limitations under the License.[39;00m
[1;32m     14[0m [38;5;66;03m# ==============================================================================[39;00m
[1;32m     15[0m [38;5;124;03m"""Keras' Distribution Strategy library."""[39;00m
[0;32m---> 18[0m [38;5;28;01mfrom[39;00m [38;5;21;01mkeras[39;00m[38;5;21;01m.[39;00m[38;5;21;01mdistribute[39;00m [38;5;28;01mimport[39;00m sidecar_evaluator

File [0;32m~/Library/Python/3.9/lib/python/site-packages/keras/distribute/sidecar_evaluator.py:17[0m
[1;32m      1[0m [38;5;66;03m# Copyright 2020 The TensorFlow Authors. All Rights Reserved.[39;00m
[1;32m      2[0m [38;5;66;03m#[39;00m
[1;32m      3[0m [38;5;66;03m# Licensed under the Apache License, Version 2.0 (the "License");[39;00m
[0;32m   (...)[0m
[1;32m     13[0m [38;5;66;03m# limitations under the License.[39;00m
[1;32m     14[0m [38;5;66;03m# ==============================================================================[39;00m
[1;32m     15[0m [38;5;124;03m"""Python module for evaluation loop."""[39;00m
[0;32m---> 17[0m [38;5;28;01mimport[39;00m [38;5;21;01mtensorflow[39;00m[38;5;21;01m.[39;00m[38;5;21;01mcompat[39;00m[38;5;21;01m.[39;00m[38;5;21;01mv2[39;00m [38;5;28;01mas[39;00m [38;5;21;01mtf[39;00m
[1;32m     19[0m [38;5;66;03m# isort: off[39;00m
[1;32m     20[0m [38;5;28;01mfrom[39;00m [38;5;21;01mtensorflow[39;00m[38;5;21;01m.[39;00m[38;5;21;01mpython[39;00m[38;5;21;01m.[39;00m[38;5;21;01mplatform[39;00m [38;5;28;01mimport[39;00m tf_logging [38;5;28;01mas[39;00m logging

[0;31mModuleNotFoundError[0m: No module named 'tensorflow'
ModuleNotFoundError: No module named 'tensorflow'

